product_filter/views.py

```diff
@@ -7,18 +7,95 @@
 from django.http import Http404
 from rest_framework.pagination import PageNumberPagination
 
+# Custom Filters: Custom filtering classes.
+class KeywordFilter:
+    def filter(self, queryset, keywords):
+        if keywords:
+            return queryset.filter(Q(name__icontains=keywords) | Q(description__icontains=keywords) | Q(category__name__icontains=keywords))
+        return queryset
+
+class CategoryFilter:
+    def filter(self, queryset, category):
+        return queryset.filter(category__name=category) if category else queryset
+
+class SubCategoryFilter:
+    def filter(self, queryset, sub_category):
+        return queryset.filter(category__productsubcategory__name=sub_category) if sub_category else queryset
+    
+class DiscountFilter:
+    def filter(self, queryset, discount):
+        return queryset.filter(discount_price=discount) if discount else queryset
+    
+class PriceFilter:
+    def filter(self, queryset, min_price, max_price):
+        return queryset.filter(Q(price__gte=min_price) & Q(price__lte=max_price)) if min_price and max_price else queryset
 
-# Create your views here.
-class FilterProductView(APIView):
-
-    # pagination_class = PageNumberPagination
-    # page_size = 10
+class RatingFilter:
+    def filter(self, queryset, rating):
+        return queryset.filter(rating_id__rating__gte=rating) if rating else queryset
     
+class HighestPriceFilter:
+    def filter(self, queryset, highest_price):
+        if highest_price:
+            if highest_price == 'true':
+                # Implement logic to order by highest price
+                return queryset.order_by('-price')
+            else:
+                # Implement logic to order by lowest price
+                return queryset.order_by('price')
+        return queryset
```

Comment:
We have to refactor this function.
What's the type of highest_price?

Reply:
yes, I can make the logic more concise

If the "highest_price" parameter is set to "true," it sorts the products from the highest price to the lowest price. 
If set to "false", it sorts them from the lowest price to the highest price. 

If "highest_price" is not provided the query set remains unsorted.